src/main/java/jp/co/seattle/library/controller/AddBooksController.java

```diff
@@ -83,15 +90,36 @@ public String insertBook(Locale locale,
                 return "addBook";
             }
         }
+        
+        List<String> list = new ArrayList<String>();
+        if(title.equals("") || author.equals("") || publisher.equals("") || publishDate.length() == 0){
+        	list.add("<p>必須項目を入力してください。</p>");
+        }
+        
+        if(!(publishDate.matches("^[0-9]{8}"))) {
+        	list.add("<p>出版日は半角数字のYYYYMMDD形式で入力してください。</p>");
+        	
+        }
+        
+        if(ISBN.length() != 0 && !(ISBN.matches("^[0-9]{10}|[0-9]{13}"))) {
+        	list.add("<p>ISBNの桁数または半角数字が正しくありません。</p>");
+        }
+        
+        if(list.size() > 0) {
+        	model.addAttribute("addErrorMessage", list);
+        	model.addAttribute("bookInfo", bookInfo);
+        	return "addBook";
+        }
 
         // 書籍情報を新規登録する
         booksService.registBook(bookInfo);
-
         model.addAttribute("resultMessage", "登録完了");
 
         // TODO 登録した書籍の詳細情報を表示するように実装
         //  詳細画面に遷移する
+        bookInfo.setBookId(booksService.getMaxId());
+        model.addAttribute("bookDetailsInfo", booksService.getBookInfo(bookInfo.getBookId()));
```

Comment:
こっちでもできますね
maxIdはbookInfoにセットしなくても問題ないです
```suggestion
        int maxId = bookInfo.setBookId(booksService.getMaxId());
        model.addAttribute("bookDetailsInfo", booksService.getBookInfo(maxId));
```